src/fields/equation_output.cc

```diff
@@ -318,12 +323,20 @@ void EquationOutput::output(TimeStep step)
     }
 
     // observe output
-    for(FieldCommon * field : this->field_list) {
-        if ( field->flags().match( FieldFlag::allow_output) ) {
-            if (observe_fields_.find(field->name()) != observe_fields_.end()) {
-                field->observe_output( observe_ptr );
+    {
+        FieldSet used_fields;
+        uint n_fields = 0;
+        for(FieldCommon * f_acc : this->field_list) {
```

Comment:
Just form the output field set here and pass it to the assembly together with Observe instance.